trollmoves/dispatcher.py

```diff
@@ -405,23 +285,114 @@ def create_dest_url(self, msg, client, conf):
         host_path = urlunsplit((parts.scheme, parts.netloc, path, parts.query, parts.fragment))
         return host_path, connection_parameters, client
 
-    def close(self):
-        """Shutdown the dispatcher."""
-        logger.info('Terminating dispatcher.')
-        self.loop = False
-        try:
-            self.listener.stop()
-        except Exception:
-            logger.exception("Couldn't stop listener.")
-        if self.publisher:
+
+class PublisherReporter:
```

Comment:
Should the base class be eventually in Posttroll?

Reply:
I don't think so, this one will be pretty specific to file moving I believe.